overlord/assertstate/assertstate.go

```diff
@@ -543,15 +557,19 @@ func ValidationSetAssertionForMonitor(st *state.State, accountID, name string, s
 // checks if it's not in conflict with existing validation sets in enforcing mode
 // (all currently tracked validation set assertions get refreshed), and if they
 // are valid for installed snaps.
-func ValidationSetAssertionForEnforce(st *state.State, accountID, name string, sequence int, userID int, snaps []*snapasserts.InstalledSnap) (vs *asserts.ValidationSet, err error) {
+func ValidationSetAssertionForEnforce(st *state.State, accountID, name string, sequence int, userID int, snaps []*snapasserts.InstalledSnap, opts *RefreshAssertionsOptions) (vs *asserts.ValidationSet, err error) {
+	if opts == nil {
+		opts = &RefreshAssertionsOptions{}
+	}
+
 	deviceCtx, err := snapstate.DevicePastSeeding(st, nil)
 	if err != nil {
 		return nil, err
 	}
 
 	// refresh all currently tracked validation set assertions (this may or may not
 	// include the one requested by the caller).
-	if err = RefreshValidationSetAssertions(st, userID); err != nil {
+	if err = RefreshValidationSetAssertions(st, userID, opts); err != nil {
```

Comment:
Shouldn't we simply pass have `opts := &RefreshAssertionsOptions{IsAutoRefresh: true}` here (and leave ValidationSetAssertionForEnforce signature intact - not take opts)?

Reply:
The idea was just to make the API consistent. I think Samuele mentioned in an earlier comment that the functions should be consistent (but I may have misunderstood/misapplied the meaning)

Reply:
Sorry, maybe I wasn't clear; I mean ValidationSetAssertionForMonitor doesn't take opts, but ValidationSetAssertionForEnforce does take them, so they are not consistent in that regard. Can we keep ValidationSetAssertionForEnfore consistent with  ValidationSetAssertionForMonitor and simply hardcode opts we pass to RefreshValidationSetAssertions here? I guess the question is when do we want to pass autorefresh:true and false here from the outside? On automatic snap refreshes vs manual installs maybe? CC @pedronis ?

Reply:
sorry, I missed this skimming the last commit. We want:

```
opts := &RefreshAssertionsOptions{IsAutoRefresh: false}
```
(false!, not true)

here afaict.  We don't plan to use ValidationSetAssertionFor* in the context of auto refresh so they don't need to take options.